src/dnx/dnx.cpp

```diff
@@ -110,125 +85,139 @@ int BootstrapperOptionValueNum(LPCTSTR pszCandidate)
     return -1;
 }
 
-void FreeExpandedCommandLineArguments(int nArgc, dnx::char_t** ppszArgv)
+size_t FindAppBaseOrNonHostOption(size_t argc, dnx::char_t**argv)
 {
-    for (int i = 0; i < nArgc; ++i)
+    for (size_t i = 0; i < argc; i++)
     {
-        delete[] ppszArgv[i];
-    }
-    delete[] ppszArgv;
-}
+        if (strings_equal_ignore_case(argv[i], _X("--appbase")))
+        {
+            return i;
+        }
 
-dnx::char_t* GetAppBaseParameterValue(int argc, dnx::char_t* argv[])
-{
-    for (auto i = 0; i < argc - 1; ++i)
-    {
-        if (dnx::utils::strings_equal_ignore_case(argv[i], _X("--appbase")))
+        auto option_num_args = BootstrapperOptionValueNum(argv[i]);
+        if (option_num_args < 0)
         {
-            return argv[i + 1];
+            return i;
         }
+
+        i += option_num_args;
     }
 
-    return nullptr;
+    return argc;
 }
 
-bool ExpandCommandLineArguments(int nArgc, dnx::char_t** ppszArgv, int& nExpandedArgc, dnx::char_t**& ppszExpandedArgv)
+dnx::char_t* GetAppBaseParameterValue(size_t argc, dnx::char_t* argv[])
 {
-    // If no args or '--appbase' is already given and it has a value
-    if (nArgc == 0 || GetAppBaseParameterValue(nArgc, ppszArgv))
+    auto index = FindAppBaseOrNonHostOption(argc, argv);
+
+    // no parameters or '--appbase' is the last value in the array or `--appbase` not found
+    if (argc == 0 || index >= argc - 1 || argv[index][0] != _X('-'))
```

Comment:
No. When `BootstrapperOptionValueNum` reaches the .dll argument it will return -1 and then we will break the loop in `FindAppBaseOrNonoHostOption` and return index to this argument. This scenario is also covered by tests - `ExpandCommandLineArguments_should_ignore_appbase_after_bootstrapper_commands` (except for we break the loop on `.` instead of the dll but it does not matter from the perspective of the `BootstrapperOptionValueNum` function since it breaks on anything it can't understand so `.` and path falls into the same category).